### PR TITLE
BugFix | #1604 | Duplicate extension key prefix

### PR DESCRIPTION
**Pull request by AlexNDRmac**
Refs:  #1604

### CODE REVIEW

**Comment by AlexNDRmac** [comment]
cc @dreamsxin Now you can see what I meant when told about fix in `Library/Compiler.php`
There is Test with Expected result, so we can use it as target point.
Travis failed buld: https://travis-ci.org/phalcon/zephir/jobs/495101242

**Comment by dreamsxin** [comment]
This config 
```json
    "globals": {
        "test.my_setting_1": {
            "type": "bool",
            "default": false
        },
    },
```
change to:
```json
    "globals": {
        "my_setting_1": {
            "type": "bool",
            "default": false
        },
    },
```
will generate `test.my_setting_1`.

File: Library/Code/Builder/Struct.php [hunk 1]
```diff
@@ -143,24 +143,32 @@ public function getCDefault($name, $global, $namespace)
     }
 
     /**
-     * @param mixed $name
-     * @param mixed $global
-     * @param mixed $namespace
+     * Process Globals for phpinfo() page.
+     *
+     * @see https://docs.zephir-lang.com/en/0.10/globals
```
Comment (sergeyklay) [inline Library/Code/Builder/Struct.php]:
```suggestion
     * @see https://docs.zephir-lang.com/latest/en/globals
```

File: Library/Code/Builder/Struct.php [hunk 2]
```diff
@@ -143,24 +143,32 @@ public function getCDefault($name, $global, $namespace)
...
+        $structName = $this->simpleName.'.'.$name;
+
         if (isset($global['ini-entry'])) {
             $iniEntry = $global['ini-entry'];
         }
-        $structName = $this->simpleName.'.'.$name;
+
         if (!isset($iniEntry['name'])) {
             $iniName = $namespace.'.'.$structName;
+        //$iniName = $structName;
```
Comment (sergeyklay) [inline Library/Code/Builder/Struct.php]:
```suggestion
```

File: config.json [hunk 3]
```diff
@@ -1,8 +1,8 @@
 {
     "namespace": "test",
-    "name": "Test Extension",
-    "description": "Description test for<br\/>Test Extension",
-    "author": "Zephir Team",
+    "name": "Zephir Test Extension",
+    "description": "<b>Zephir</b> - is a high level <strong>Programming Language</strong> that eases the creation and maintainability of extensions for PHP.",
```
Comment (sergeyklay) [inline config.json]:
```suggestion
    "description": "Description test for<br\/>Test Extension",
```

Comment (sergeyklay) [reply config.json]: It is not Zephir. It is test extension

File: ext/php_test.h [hunk 4]
```diff
@@ -10,19 +10,28 @@
...
 
-#define PHP_TEST_NAME        "Test Extension"
+#define PHP_TEST_NAME        "Zephir Test Extension"
 #define PHP_TEST_VERSION     "1.0.0"
 #define PHP_TEST_EXTNAME     "test"
-#define PHP_TEST_AUTHOR      "Zephir Team"
+#define PHP_TEST_AUTHOR      "Zephir Team and contributors"
 #define PHP_TEST_ZEPVERSION  "0.11.9-$Id$"
-#define PHP_TEST_DESCRIPTION "Description test for<br/>Test Extension"
+#define PHP_TEST_DESCRIPTION "<b>Zephir</b> - is a high level <strong>Programming Language</strong> that eases the creation and maintainability of extensions for PHP."
```
Comment (sergeyklay) [inline ext/php_test.h]:
```suggestion#define PHP_TEST_DESCRIPTION "Description test for<br/>Test Extension"
```

It is not Zephir. It is test extension

File: ext/php_test.h [hunk 5]
```diff
@@ -10,19 +10,28 @@
 
 #include "kernel/globals.h"
 
-#define PHP_TEST_NAME        "Test Extension"
+#define PHP_TEST_NAME        "Zephir Test Extension"
```
Comment (sergeyklay) [inline ext/php_test.h]:
```suggestion
#define PHP_TEST_NAME        "Test Extension"
```

File: config.json [hunk 6]
```diff
@@ -1,8 +1,8 @@
 {
     "namespace": "test",
-    "name": "Test Extension",
-    "description": "Description test for<br\/>Test Extension",
-    "author": "Zephir Team",
+    "name": "Zephir Test Extension",
```
Comment (sergeyklay) [inline config.json]:
```suggestion
    "name": "Test Extension",
```

**Review by sergeyklay — CHANGES_REQUESTED** [review]

File: config.json (hunk 6 above)
Comment (AlexNDRmac) [reply config.json]: Everything fixed

**Comment by AlexNDRmac** [comment]
@dreamsxin you were right :)  I did't do the tests correctly. Thank you for your participation.

**Review by sergeyklay — APPROVED** [review]

**Comment by sergeyklay** [comment]
Thank you

**Pull request merged**